src/main.cpp

```diff
@@ -3087,8 +3087,14 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
         //   to ConnectBlock() (and thus never on the main chain) will stay with
         //   these set to null.
         if (chainparams.GetConsensus().NetworkUpgradeActive(pindex->nHeight, Consensus::UPGRADE_NU5)) {
+            try {
+                pindex->hashFinalOrchardRoot = orchard_tree.root();
+            } catch (const OrchardAnchorBottom& e) {
+                return state.DoS(100,
+                    error("ConnectBlock(): derived the bottom value computing the root of the Orchard commitment tree"),
+                    REJECT_INVALID, "bad-orchard-root-in-block");
+            }
             pindex->hashAuthDataRoot = hashAuthDataRoot.value();
-            pindex->hashFinalOrchardRoot = orchard_tree.root(),
```

Comment:
Revert this change once the check is moved (since we will be guaranteed that this succeeds).